### ISSUE TITLE
Quest: find electric car chargers

### ISSUE THREAD

**Issue by gappleto97** [westnordost/StreetComplete#117]
This would be a good prompt to put at parking lots. It would be even more helpful to label what type it had. So if you wanted to borrow from the road surface UI, it would be:

* no
* yes (AC)
    * J1772
    * wall outlet
* yes (DC)
    * CHAdeMO
    * Tesla
* yes (mixed)

**Comment by matkoniecz** [comment]
It should be smarter than asking this on the every parking lot. In my region (Poland) electric car chargers are really rare, marking every parking as missing electric charger would be obnoxious.

**Comment by matkoniecz** [comment]
AKA  Feature Request: Filter quests #35

**Comment by lightonflux** [comment]
I also have a problem with "no". It would get overwhelming and almost no one would enable it (without OCD).

But it would be nice if existing chargers are enhanced. Do you know how well existing chargers are tagged?

→ [Combinations](https://taginfo.openstreetmap.org/tags/amenity=charging_station#combinations])

**Comment by westnordost** [comment]
Yes, quests for actively finding something will not be added (in the medium term), but quests for enhancing something will.

**Comment by rugk** [comment]
So is it planed to add [the socket type](https://wiki.openstreetmap.org/wiki/Key:socket) and/or the [capacity](https://wiki.openstreetmap.org/wiki/Key:capacity)?`Socket type may require expert knowledge, unless it is written out somewhere… (but that may be likely, I have no idea.)

**Comment by gappleto97** [comment]
In my experience, socket type is not advertised, but you're very aware of what kinds your car can accept. Pretty much every EV owner can recognize a J1772 cord, though many might not be able to name it.

Another large chunk can recognize a CHaDEMO cord (caps wrong?), but usually as "that common-ish one my car doesn't accept".

**Comment by westnordost** [comment]
Okay, I fear for now this quest can't be added:

The plug type must be recognizable by anyone (not an electric car driver). I checked such a station in Hamburg and it doesn't say anywhere at all which socket type. You had to pay to even reveal the socket. So it's not solveable by a passerby.
Additionally, the fact that no hint for the socket standard was given on the station anywhere hints to the fact that it is standardised at least regionally. (J1772 seems to be the global standard and cars that have something else are usually backward-compatible to that socket.)
So, that means it also is not so valuable to record that data.

Which socket type etc is supported by a station should be derivable either from the country / region or at least from the operator of the station.

In short: Will close this, you can open another for charging station operator if you wish.

**Comment by WolfgangFahl** [comment]
> Yes, quests for actively finding something will not be added (in the medium term), but quests for enhancing something will.
I think this should be reopened. The need fo adding ne amenity charging_station information is imment for the near future.  I'll have to test the current state of the app to be more specific about the need and get back.

**Comment by matkoniecz** [comment]
> I think this should be reopened. The need fo adding ne amenity charging_station information is imment for the near future.

This does not mean that it is within scope of this program.

>  I'll have to test the current state of the app to be more specific about the need and get back.

I strongly recommend reading also https://github.com/westnordost/StreetComplete/wiki/Adding-new-Quests-to-StreetComplete

**Comment by WolfgangFahl** [comment]
Thx for looking into this 

> Only existing elements can be extended, no elements can be added or removed.
It would be great to add elements. In the case of charging stations I think:
- a photo
- the GPS position
- the QR (codes)
of the charging station should be gathered. I think this pattern hold true for most items to be added 
photo, position and url are the starting points for POIs.  Is there any other app or issue that focuses on this?

**Comment by rugk** [comment]
Wait a second. What QR codes do you talk about? It's totally new to me, that these stations have QR codes? And if so, is it just one provider, or do really _all of them_ have QR codes? (And if so, what is encoded in these QR codes?)

**Comment by WolfgangFahl** [comment]
In Germany https://de.wikipedia.org/wiki/Lades%C3%A4ulenverordnung as of mid 2017 has made QR codes practically mandatory. New charging stations would only be able to avoid them if they have some other means of "spontaneous payment option".

**Comment by rugk** [comment]
Okay, that is interesting. Unfortunately, the Wikipedia article itself does not mention the QR codes. Also the whole thing is still quite vague ("[they] mostly point to…", "practically mandatory"). So we need more details there… (also data format of QR codes, or is it just a link etc.?)

I mean, _if_ there is some useful data in the QR code, it would likely be "easy"™ to make a quest to collect this and add to OSM. However, what data is in there even? An "id for the charging station" is hardly useful for OSM… the "payment system" could be a useful information (actually, something like payment provider or generally the car charger provider or so).

**Comment by gappleto97** [comment]
The only ones I have ever seen with QR codes are the ones from GE, and they suck. AFAIK, no major US charging network uses QR codes. They are *all* NFC based.

**Comment by Cj-Malone** [comment]
I know this is closed and old, but I just wanted to confirm. I surveyed 2 near me in England, neither had the type of socket on any visible documentation. I could open the flap, and so I guess I could compare it against a picture, if this was ever added to StreetComplete it should probably be off by default. I also got 'capacity' really easily, so that could be on by default.

**Comment by stefan005** [comment]
Not only for cars but also for e-bikes the information which socket a charging station has is interesting.
I personally would love to see the option to add the **capacity**=int and **socket_type:***=int tags within streetcomplete.
At many charging stations I know, it's easy to identify what socket type it has. (At least type2 and schuko are easy to identify for me). If needed, it would be ok for me to make this an optional quest, because it might not be easy to identify at some charging stations.
I would also be interested in the tags **motorcars**=yes/no and **bicycle**=yes/no

I'm willing to implement this quest and create a pull request. (excluding artwork)
Please let me know whether I should put some time and effort into implementing some of the tags. I would highly appreciate to have the tags mentioned above included in the app!

**Comment by matkoniecz** [comment]
See https://github.com/westnordost/StreetComplete/issues/117#issuecomment-325938671 - at least in some cases socket_type is untaggable. Is there maybe some tag for "socket type is hidden"?

`capacity`, `motorcars`, `bicycle` - if you will get no reply here (what is likely it is a closed issue) and the quest matches requirements that opening a new issue is likely a good idea (is it something that was proposed already and has its own issue? Then comment there.)

**Comment by rugk** [comment]
Yep, better ask new ideas for new issues (_if_ they meet the requirements etc.) in a _new_ issue.
